api/python/provisioner/commands/configure_setup.py

```diff
@@ -90,145 +101,51 @@ def __attrs_post_init__(self):
-    hostname: str = NodeNetworkParams.hostname
-    roles: List = NodeNetworkParams.roles
-    data_public_interfaces: List = NodeNetworkParams.data_public_interfaces
-    data_private_interfaces: List = NodeNetworkParams.data_private_interfaces
-    data_public_ip: str = NodeNetworkParams.data_public_ip
-    data_netmask: str = NodeNetworkParams.data_netmask
-    data_gateway: str = NodeNetworkParams.data_gateway
-    mgmt_interfaces: List = NodeNetworkParams.mgmt_interfaces
-    mgmt_public_ip: str = NodeNetworkParams.mgmt_public_ip
-    mgmt_netmask: str = NodeNetworkParams.mgmt_netmask
-    mgmt_gateway: str = NodeNetworkParams.mgmt_gateway
-    data_private_ip: str = NodeNetworkParams.data_private_ip
-    bmc_user: str = NodeNetworkParams.bmc_user
-    bmc_secret: str = NodeNetworkParams.bmc_secret
-
-    _optional_param = [
-        'data_public_ip',
-        'roles',
-        'data_netmask',
-        'data_gateway',
-        'data_private_ip',
-        'mgmt_interfaces',
-        'mgmt_public_ip',
-        'mgmt_netmask',
-        'mgmt_gateway'
-    ]
-
-    def __attrs_post_init__(self):
-        params = attr.asdict(self)
-        missing_params = []
-        for param, value in params.items():
-            if value == UNCHANGED and param not in self._optional_param:
-                missing_params.append(param)
-        if len(missing_params) > 0:
-            raise ValueError(f"Mandatory param missing {missing_params}")
-
-
 @attr.s(auto_attribs=True)
 class ConfigureSetup(CommandParserFillerMixin):
     input_type: Type[inputs.NoParams] = inputs.NoParams
     _run_args_type = RunArgsConfigureSetup
 
-    validate_map = {
-        "cluster": NetworkParamsValidation,
-        "node": NodeParamsValidation,
-        "storage": StorageEnclosureParamsValidation
-    }
+    validate_map = {f'{CLUSTER}': NetworkParamsValidation,
+                    f'{NODE}': NodeParamsValidation,
+                    f'{STORAGE_ENCLOSURE}': StorageNodeParamsValidation,
+                    f'{NODE_DEFAULT}': NodeNetworkParamsValidation,
+                    f'{STORAGE_DEFAULT}': StorageEnclosureParamsValidation}
 
-    def _parse_params(self, input):
+    @staticmethod
+    def _parse_params(input):
         params = {}
         for key in input:
-            logger.debug(f"Key being processed: {key}")
             val = key.split(".")
-            if len(val) > 1 and val[-1] in [
-                'ip', 'user', 'secret', 'type', 'interfaces',
-                'private_interfaces', 'public_interfaces',
-                'gateway', 'netmask', 'public_ip', 'private_ip'
-            ]:
+            conf_values = [
+                'ip', 'user', 'secret', 'ipaddr', 'interfaces', 'gateway',
+                'netmask', 'public_ip_addr', 'type',
+                'id', 'roles', 'data_devices', 'metadata_device']
+
+            if len(val) > 2 and val[-1] in conf_values:
+                params[f'{val[-3]}_{val[-2]}_{val[-1]}'] = input[key]
```

Comment:
I think we don't need above code block.

` if len(val) > 2 and val[-1] in conf_values:
                params[f'{val[-3]}_{val[-2]}_{val[-1]}'] = input[key]`

logic is to  append only last 2 elements 
so that we can uniquely identify each element.
example:
 network.data.netmask=
network.data.gateway=
network.mgmt.netmask=

we need only below data which is params for validation class.
data_netmast
data.gateway
mgmt.netmask

@divya-vijay  can you please check if code works fine after removing above block.

Reply:
In latest naming changes of `config.ini`, the length of few variables was more than 2. 
Updated parsing logic and removed this block now. Thanks !